test/python/circuit/library/test_blueprintcircuit.py

```diff
@@ -209,6 +209,34 @@ def _build(self):
             math.pi / 2,
         )
 
+    def test_copy_empty_like(self):
+        """Test copying empty-like.
+
+        Regression test of #13535.
+        """
+        circuit = EfficientSU2(2)
+        circuit.global_phase = -2
+        circuit.metadata = {"my_legacy": "i was a blueprintcircuit once"}
+
+        cpy = circuit.copy_empty_like()
+        expected = QuantumCircuit(
+            circuit.num_qubits, global_phase=circuit.global_phase, metadata=circuit.metadata
+        )
+        self.assertEqual(cpy, expected)
+        self.assertNotIsInstance(cpy, BlueprintCircuit)
+
+    def test_copy_empty_like_post_build(self):
+        """Test copying empty-like after building the circuit."""
+        circuit = EfficientSU2(2)
+        _ = circuit.count_ops()  # trigger building the circuit
+
+        cpy = circuit.copy_empty_like()
+
+        circuit.num_qubits = 3  # change the original circuit
+
+        expected = QuantumCircuit(2)  # we still expect an empty 2-qubit circuit
+        self.assertEqual(cpy, expected)
```

Comment:
It's good to test mutating `circuit` to ensure no modifications sneak through in shared state, but it's the `_build` method getting called on `cpy` which would be the full test, I think?

Reply:
There should be no `_build` on `cpy` since that's just a plain `QuantumCircuit` 🙂

Reply:
Yeah I know, because you've fixed it, but the point is that this test doesn't _necessarily_ test the regression behaviour, because there's no code in here that super obviously would have caused a build if your PR didn't actually fix the problem.  `__eq__` _might_ do it accidentally, but you could easily imagine an optimisation in `BlueprintCircuit` that would cause it not to.

Might be good to do something to `cpy` which is API guaranteed to cause a build _if it were_ a `BlueprintCircuit` before testing for emptiness.